.github/workflows/build.yaml

```diff
@@ -7,12 +7,12 @@ on:
     types: [released, prereleased]
   workflow_dispatch:
 
-permissions:
-  contents: read
-  id-token: write
-
 jobs:
   pip:
+    environment: pypi
+    permissions:
+      contents: read
+      id-token: write
```

Comment:
Could you please explain this change please?

Reply:
> [_only set the `id-token: write` permission in the job that does publishing, not globally_](https://github.com/pypa/gh-action-pypi-publish/blob/a712d989cca7b5d2b239f655efb2512082d5bce8/README.md?plain=1#L81-L86)

Albeit it's arguably pointless in a single-job workflow, the recommendation is to set the permissions at the job level rather than globally.

TL;DR another unnecessary change you may refuse within reason.

Reply:
Are those permissions merged? If yes, then I'd prefer `contents: read` to be on the top-level and `id-write` on jobs level. Otherwise, I'm okay with your changes. :)

Reply:
> _Are those permissions merged?_

They aren't, unfortunately. The `permissions` block on jobs overrides the global one.

Reply:
Okay, I added `contents: read` back at the top-level so that it'll be used if we add additional jobs here.